web/src/confirm_dialog.ts

```diff
@@ -1,12 +1,17 @@
+import $ from "jquery";
+
 import * as dialog_widget from "./dialog_widget";
 import type {DialogWidgetConfig} from "./dialog_widget";
 import {$t_html} from "./i18n";
 
 export function launch(conf: DialogWidgetConfig): string {
-    return dialog_widget.launch({
-        close_on_submit: true,
-        focus_submit_on_open: true,
-        html_submit_button: $t_html({defaultMessage: "Confirm"}),
-        ...conf,
-    });
+    if ($(".modal__container[role='dialog']").length === 0) {
+        return dialog_widget.launch({
+            close_on_submit: true,
+            focus_submit_on_open: true,
+            html_submit_button: $t_html({defaultMessage: "Confirm"}),
+            ...conf,
+        });
+    }
+    return "Dialog already open";
```

Comment:
It seems like returning this new value might be problematic for some callers?

Reply:
Maybe, `mark_all_as_read` is the only function the expect `launch` to return a modal_id, even then it shouldn't be a problem. but future uses of `confirm_modal` might require it. What would `modal_id` should be returned if no new modal is created?